src/app/search/search-filters/dupes.tsx

```diff
@@ -230,27 +231,46 @@ function computeStatDupeLower(
   const statsCache = new Map<DimItem, number[]>();
   const dupes = new Set<string>();
 
+  // Run through all the armor once to get their stats, so we don't have to
+  // compute item => stats more than once
   for (const item of armor) {
     if (item.stats && item.power && item.bucket.hash !== BucketHashes.ClassArmor) {
       const statsToConsider = customStats[item.classType] ?? armorStats;
       statsCache.set(
         item,
-        _.sortBy(
-          item.stats.filter((s) => statsToConsider.includes(s.statHash)),
-          (s) => s.statHash
-        ).map((s) => s.base)
+        item.stats
+          .filter((s) => statsToConsider.includes(s.statHash))
+          .sort((a, b) => a.statHash - b.statHash)
+          .map((s) => s.base)
       );
     }
   }
 
+  // For each group of items that should be compared against each other
   for (const group of grouped) {
     const statSet = new StatsSet<DimItem>();
+    // Add a mapping from stats => item to the statsSet for each item in the group
     for (const item of group) {
       const stats = statsCache.get(item);
       if (stats) {
-        statSet.insert(stats, item);
+        if (isArtifice(item)) {
+          // Artifice armor can be +3 in any one stat, so we insert a separate
+          // version of the stats for each stat considered
+          const statsToConsider = customStats[item.classType] ?? armorStats;
+          for (let i = 0; i < statsToConsider.length; i++) {
+            const modifiedStats = [...stats];
+            // One stat gets +3
+            modifiedStats[i] += 3;
+            statSet.insert(modifiedStats, item);
+          }
+        } else {
+          statSet.insert(stats, item);
+        }
       }
     }
+
+    // Now run through the items in the group again, checking against the fully
+    // populated stats set to see if there's something better
```

Comment:
...and yes, there is always something better for artifice pieces since this compares each artifice items' base stats against six copies of it, each with a boosted stat. Meaning this highlights every artifice piece I have.

Reply:
Good point, maybe I should pay full attention to this

Reply:
OK, take a whack at this - one last try before I fold and write tests...